.github/workflows/e2e_core.yml

```diff
@@ -59,6 +59,16 @@ jobs:
       any_changed: ${{ steps.get-changed.outputs.any_changed }}
     steps:
     - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+    - name: Fetch PR's merge commit
```

Comment:
Do we have "approve and run" button for `pull_request_target` trigger? If not, we can't blindly fetch content from users in here, especially since it runs on our sel-hosted runners

Reply:
Given it shows up in the check UI I would expect the *approval and run* button also applies but I don't have proof for that.

The more caveats and gotchas turn up the more I'm leaning towards going with #1734 for now.

Reply:
> Given it shows up in the check UI I would expect the _approval and run_ button also applies but I don't have proof for that.

I just found in [other docs](https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks) that it's not true, unfortunately:
> Workflows triggered by pull_request_target events are run in the context of the base branch. Since the base branch is considered trusted, workflows triggered by these events will always run, regardless of approval settings.

Reply:
> The more caveats and gotchas turn up the more I'm leaning towards going with #1734 for now.

yeah... either this or we come back to idea with `dispatch` trigger (like benchmarks)